tools/c7n_azure/c7n_azure/handler.py

```diff
@@ -64,10 +65,9 @@ def run(event, context, subscription_id=None):
         for p in policies:
             try:
                 p.push(event, context)
-            except (CloudError, AzureHttpError) as error:
+            except Exception as error:
```

Comment:
I think it is better if we catch the known/expected cloud based exceptions and let other ones actually propagate up and fail custodian.

Reply:
If there are multiple policies, we still have to run through all of them.. So, what do you think about some flag `had exception` and throwing generic `custodian policy failed` exception if at least one policy failed?

Reply:
I like the idea of throwing if 1+ policy fails

Reply:
this would be better as an try/except/finally, with the finally doing a reset_session_cache